src/applications/apps.js

```diff
@@ -105,18 +105,35 @@ export function getAppsToMount() {
     .filter(shouldBeActive)
 }
 
+export function unregisterApplication(appName) {
```

Comment:
I am conflicted about adding this API. The motivation for it is that single-spa-playground would use it -- It allows for you to modify your single-spa application registration without reloading the page. But I don't really see a strong use case for that outside of single-spa-playground.

What you think?

Reply:
hm, personally I think it's ok to expose this. Don't have strong feelings either way

Reply:
my concerns were about exposing something that would only result in confusion, or continuing to add to our already somewhat large api surface. But I think this one is maybe okay, too, idk. @TheMcMurder @blittle thoughts?

Reply:
Yeah, I don't think you'd want people to add all sorts of custom logic, like listening for routing events to register/unregister apps. The presence of the API might confuse people thinking, "ahh, so this is how you dynamically unload and unload frameworks"

Reply:
Yeah I agree there's a chance of confusion. But I also like what Justin pointed out -- that this gives people a way to move applications out of SKIP_BECAUSE_BROKEN status and do something with them.

I think I'm overall in favor of adding this API. Thoughts?

Reply:
I also thinks it makes sense to add it. sooner or later someone will find another use case for it.

Reply:
@bboydflo I don't think we should add it for that reason. We've helped a lot of people implement single spa and AFAIK no one has ever needed to unregister an application.

The debate really centralizes on confusion of a new API when a significant part of helping others implement is confusion around our already large API, a fundamental architecture shift, and some bad examples that we're trying to move away from but people just keep finding them.

I can go either way in this still. I think it's necessary for the toiling we want to build but I don't really want to expose it to everyone else.

Reply:
I've updated this so that it's no longer exposed for normal users, but only exposed through the __SINGLE_SPA_DEVTOOLS__ global variable

Reply:
If there are use cases that pop up for single-spa users, we can revisit exporting it as a real single-spa API.

Reply:
@joeldenning I might have a use case for this .... we are building a "dashboard framework" were each widget is a small single-spa application, when a user add a widget I want to register and mount the widget, when a user removes it, I want to unmount and unregister the application. The same widget can be added multiple times into a dashboard layout so we are register applications dynamically using "random names"  but use the same "app" function which return the module loaded with SystemJS.

Reply:
Applications are generally route-based and declarative - the browser URL should determine whether they are active or not. How are you implementing the activity function and controlling dom order for these widgets? Can you express which widgets are active via a URL? If not, why are you using applications instead of parcels?

Reply:
@joeldenning thanks for the fast response, and well yes now that you point it out, doh! parcels looks like a better solution!  we will give it a try. thanks again.